src/list.tsx

```diff
@@ -258,107 +259,93 @@ type ControlProps = {
-    href?: undefined;
-    onPress?: undefined;
-    to?: undefined;
-    checkbox?: undefined;
-    switch?: undefined;
-    radioValue?: undefined;
-    newTab?: undefined;
-    fullPageOnWebView?: undefined;
+interface PressableProps {
+    onPress?: () => void;
+}
+
+interface RightyRowProps {
     right?: Right;
 }
 
-interface SwitchRowContentProps extends CommonProps {
-    href?: undefined;
-    onPress?: () => void;
-    to?: undefined;
-    right?: undefined;
-    checkbox?: undefined;
-    radioValue?: undefined;
-    newTab?: undefined;
-    fullPageOnWebView?: undefined;
+interface BasicRowContentProps extends RightyRowProps, CommonProps {}
 
+interface SwitchRowContentProps extends PressableProps, CommonProps {
     switch: ControlProps;
 }
 
-interface CheckboxRowContentProps extends CommonProps {
-    href?: undefined;
-    onPress?: () => void;
-    to?: undefined;
-    right?: undefined;
-    switch?: undefined;
-    radioValue?: undefined;
-    newTab?: undefined;
-    fullPageOnWebView?: undefined;
-
+interface CheckboxRowContentProps extends PressableProps, CommonProps {
     checkbox: ControlProps;
 }
 
 interface RadioRowContentProps extends CommonProps {
-    href?: undefined;
-    onPress?: undefined;
-    to?: undefined;
-    right?: undefined;
-    switch?: undefined;
-    checkbox?: undefined;
-    newTab?: undefined;
-    fullPageOnWebView?: undefined;
-
     radioValue: string;
 }
 
-interface HrefRowContentProps extends CommonProps {
-    checkbox?: undefined;
-    switch?: undefined;
-    radioValue?: undefined;
-    fullPageOnWebView?: undefined;
-
+interface HrefRowContentProps extends RightyRowProps, CommonProps {
     trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
     href: string;
     newTab?: boolean;
-    onPress?: undefined;
-    to?: undefined;
-    right?: Right;
 }
 
-interface ToRowContentProps extends CommonProps {
-    checkbox?: undefined;
-    switch?: undefined;
-    radioValue?: undefined;
-    newTab?: undefined;
-
+interface ToRowContentProps extends RightyRowProps, CommonProps {
     trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
     to: string;
     fullPageOnWebView?: boolean;
     replace?: boolean;
-    href?: undefined;
-    onPress?: undefined;
-    right?: Right;
 }
 
-interface OnPressRowContentProps extends CommonProps {
-    checkbox?: undefined;
-    switch?: undefined;
-    radioValue?: undefined;
-    fullPageOnWebView?: undefined;
-
+interface OnPressRowContentProps extends PressableProps, RightyRowProps, CommonProps {
     trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
     onPress: () => void;
-    href?: undefined;
-    to?: undefined;
-    right?: Right;
 }
 
-type RowContentProps =
+type RowContentProps = ExclusifyUnion<
+    | (Omit<SwitchRowContentProps, 'switch'> & {switch: ControlProps | undefined})
+    | (Omit<CheckboxRowContentProps, 'checkbox'> & {checkbox: ControlProps | undefined})
```

Comment:
can you explain this? why is the Omit needed if you're re-defining the `switch`/`checkbox` prop with the same type? I don't get it